### PR TITLE
52222 - extract zip and create iso and sha1 files

### PR DESCRIPTION
**Pull request by sumod13943**
Below functionalities are added:-

- Extract zip files.
- Create ISO files from the extracted data..
- Prepare ISO files with a sha1 signature.
- Unit Tests

### CODE REVIEW

**Review by harshal11869 — CHANGES_REQUESTED** [review]

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Utilities/FileUtility.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        {
+            var iso = new CDBuilder
+            {
+                UseJoliet = true,
+                VolumeIdentifier = "FullAVCSExchangeSet"
+            };
+
+            foreach (string? file in srcFiles)
+            {
+                var fi = new FileInfo(file);
```
[the remark this answers is not in the document]
Comment (sumod13943) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Utilities/FileUtility.cs]: Changed

**Review by harshal11869 — APPROVED** [review]
LGTM

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Utilities/FileUtility.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using System.Security.Cryptography;
+using DiscUtils.Iso9660;
+namespace UKHO.PeriodicOutputService.Common.Utility
```
Comment (ajasonca) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Utilities/FileUtility.cs:4]: Namespace shoud be UKHO.PeriodicOutputService.Common.**Utilities** to correspons to the file location.

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Utilities/FileUtility.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+            foreach (string? file in srcFiles)
+            {
+                var fileInfo = new FileInfo(file);
+                if (fileInfo.Directory.Name == directoryPath)
+                {
+                    iso.AddFile($"{fileInfo.Name}", fileInfo.FullName);
+                    continue;
+                }
+                string? srcDir = fileInfo.Directory.FullName.Replace(directoryPath, "").TrimEnd('\\');
```
Comment (ajasonca) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Utilities/FileUtility.cs:25]: srcDir doesn't need to be nullable, so you can remove the ?

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Utilities/IFileUtility.cs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+﻿namespace UKHO.PeriodicOutputService.Common.Utility
```
Comment (ajasonca) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Utilities/IFileUtility.cs:1]: Namespace should be UKHO.PeriodicOutputService.Common.Utilities to correspons to the file location.

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Fulfilment.UnitTests/Services/FulfilmentDataServiceTests.cs [hunk 5]
```diff
@@ -20,6 +21,7 @@ public class FulfilmentDataServiceTests
         private ILogger<FulfilmentDataService> _fakeLogger;
         private IFileSystemHelper _fakefileSystemHelper;
         private IConfiguration _fakeconfiguration;
+        private IFileInfo _fakeFileInfo;
 
         public FleetMangerGetAuthTokenResponseModel jwtauthUnpToken = new();
         public FleetMangerGetAuthTokenResponseModel jwtAuthJwtToken = new();
```
Comment (ajasonca) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Fulfilment.UnitTests/Services/FulfilmentDataServiceTests.cs:27]: Is variable jwtAuthJwtToken used? If not please remove.

**Review by ajasonca — CHANGES_REQUESTED** [review]
Just  a few points to address.

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Utilities/FileUtility.cs (hunk 2 above)
Comment (sumod13943) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Utilities/FileUtility.cs]: Renamed

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Utilities/FileUtility.cs (hunk 3 above)
Comment (sumod13943) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Utilities/FileUtility.cs]: Removed nullable

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Utilities/IFileUtility.cs (hunk 4 above)
Comment (sumod13943) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Utilities/IFileUtility.cs]: Renamed

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Fulfilment.UnitTests/Services/FulfilmentDataServiceTests.cs (hunk 5 above)
Comment (sumod13943) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Fulfilment.UnitTests/Services/FulfilmentDataServiceTests.cs]: Removed

**Review by ajasonca — APPROVED** [review]
Looks good thanks.

**Pull request merged**